src/DFAMin.hs

```diff
@@ -25,7 +24,7 @@ import qualified Data.List as List
 --      choose and remove a set A from Q
 --      for each c in ∑ do
 --           let X be the set of states for which a transition on c leads to a state in A
---           for each set Y in P for which X ∩ Y is nonempty do
+--           for each set Y in P for which X ∩ Y is nonempty and Y \ X is nonempty do
```

Comment:
Could we factor out a proposition:
```
Y contradicts X := Y ∩ X is nonempty and Y \ X is nonempty 
```
And use it below? I think it would be good to do `Y ∩ X` not ` X ∩ Y` everywhere too.

I am a maintainer because my interest in meta-programming, not my non-existent deep knowledge of the theory of Autonoma. I am just learning about this algorithm now and so trying to make these docs extra clear.

Reply:
I think `Y contradicts X` is more like `X ∩ Y is empty` rather than `Y ∩ X is nonempty and Y \ X is nonempty `

I'm also not a expert, so feel free :)

Reply:
I think it's better to give a name to `Y ∩ X` and `Y \ X`
i.e.
```
Y1 = Y ∩ X
Y2 = Y \ X
```

Reply:
I think the basic idea is that the intersections of the preimages collectively induce the minimal equivalence relation. However, since the primages are defined in terms of "states so far" that alone would be an impredicative / non-closed form definition. So the algorithm works to find a fixed point monotonically instead.

OK that's a lot of jargon. Basically, I call it "contradicts" because if the set contains some things that transition on "c" and some things that don't, then it must be too coarse. In opposite terms, the "true" equivalence relational respects all such preimage partitions in that no equivalence class is partitioned.

> I think it's better to give a name to Y ∩ X and Y \ X

Sure. To go full Haskell, I would do:

```haskell
contradicts :: NonEmptySet k -> Set k -> Maybe (NonEmptySet k, NonEmptySet k)
```
But I didn't want to impose upon you coming up with algol-pseudo-code-ish pattern matching :D. And I also wanted to make it not diverge too much from the Wikipedia entry :).

Reply:
what about
`X refines Y into Y1 and Y2`
the term "refine" is used in wikipedia and the references in wikipedia.

Reply:
Sure!